openff/toolkit/utils/exceptions.py

```diff
@@ -71,45 +75,218 @@ class LicenseError(ToolkitUnavailableException):
 
 
-class InvalidIUPACNameError(MessageException):
+class InvalidIUPACNameError(OpenFFToolkitException):
     """Failed to parse IUPAC name"""
 
 
-class AntechamberNotFoundError(MessageException):
+class AntechamberNotFoundError(OpenFFToolkitException):
     """The antechamber executable was not found"""
 
 
-class SMILESParseError(MessageException, ValueError):
-    """The record couple not be parsed into the given format"""
+class SMILESParseError(ValueError):
+    """The record could not be parsed into the given format"""
+
+
+class InvalidConformerError(Exception):
+    """
+    This error is raised when the conformer added to the molecule
+    has a different connectivity to that already defined.
+    or any other conformer related issues.
+    """
+
+
+class SmilesParsingError(Exception):
+    """
+    This error is raised when parsing a SMILES string results in an error.
+    """
+
+
+class NotAttachedToMoleculeError(OpenFFToolkitException):
+    """Exception for when a component does not belong to a Molecule object, but is queried"""
+
+
+class DuplicateUniqueMoleculeError(OpenFFToolkitException):
+    """
+    Exception for when the user provides indistinguishable unique molecules when trying to identify atoms from a PDB
+    """
+
+
+class NotBondedError(OpenFFToolkitException):
+    """
+    Exception for when a function requires a bond between two atoms, but none is present
+    """
+
+
+class InvalidBoxVectorsError(OpenFFToolkitException):
+    """
+    Exception for setting invalid box vectors
+    """
+
+
+class InvalidPeriodicityError(OpenFFToolkitException):
+    """
+    Exception for setting invalid periodicity
+    """
+
+
+class MissingUniqueMoleculesError(OpenFFToolkitException):
+    """
+    Exception for when unique_molecules is required but not found
+    """
+
+
+class SMIRKSMismatchError(OpenFFToolkitException):
+    """
+    Exception for cases where smirks are inappropriate
+    for the environment type they are being parsed into
+    """
+
+
+class SMIRKSParsingError(OpenFFToolkitException):
+    """
+    Exception for when SMIRKS are not parseable for any environment
+    """
+
+
+class ParameterHandlerRegistrationError(OpenFFToolkitException):
+    """
+    Exception for errors in ParameterHandler registration
+    """
+
+
+class SMIRNOFFVersionError(OpenFFToolkitException):
+    """
+    Exception thrown when an incompatible SMIRNOFF version data structure is attempted to be read.
+    """
+
+
+class SMIRNOFFAromaticityError(OpenFFToolkitException):
+    """
+    Exception thrown when an incompatible SMIRNOFF aromaticity model is checked for compatibility.
+    """
+
+
+class _DeprecatedParseError(Exception):
+    """DEPRECATED: Error for when a SMIRNOFF data structure is not parseable by a ForceField"""
+
+    def __init__(self, msg):
+        super().__init__(msg)
+        self.msg = msg
+
+    def __str__(self):
+        return self.msg
```

Comment:
I'd just inherit from _DeprecatedMessageException. As noted in https://github.com/openforcefield/openff-toolkit/pull/1021/files#r687269171, this will hopefully get rid of that LGTM alert too.

```suggestion
class _DeprecatedParseError(_DeprecatedMessageException):
    """DEPRECATED: Error for when a SMIRNOFF data structure is not parseable by a ForceField"""
```